Jenkinsfile

```diff
@@ -343,6 +350,53 @@ pipeline {
                 githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
               }
             }
+            stage('Benchmarks') {
+              // agent { label 'metal' }
+              // TODO: tools are only installed in this worker for now
+              agent { label 'worker-854309' }
+              environment {
+                REPORT_FILE = 'apm-agent-benchmark-results.json'
+                HOME = "${env.WORKSPACE}"
+              }
+              when {
+                beforeAgent true
+                allOf {
+                  anyOf {
+                    branch 'master'
+                    tag pattern: 'v\\d+\\.\\d+\\.\\d+.*', comparator: 'REGEXP'
+                    expression { return params.Run_As_Master_Branch }
+                    // TODO: It's required to configure the benchmark for dotnet to be dryRun
+                    // or prepare an ES where to send the data to
+                    // expression { return env.BENCHMARK_UPDATED != "false" }
+                  }
+                  expression { return env.ONLY_DOCS == "false" }
+                }
+              }
+              options {
+                warnError('Benchmark failed')
```

Comment:
Let's avoid a build failure if the benchmark stage failed for whatever reason, it will report the build as unstable rather than a failure, in other words, it will keep running the next stages as usual.